### PR TITLE
WIP: Split into subpackages

### PR DESCRIPTION
**Pull request by cgwalters**
- Some of our hacks (e.g. the systemd coredump bit) are
   maipo-specific
 - We really want to clearly distinguish these things; separate
   `-release` packages helps, but even better is having different
   version numbers in `/etc/os-release`.

### CODE REVIEW

**Comment by cgwalters** [comment]
This encodes the `47` for maipo and `48` for ootpa.

File: redhat-release-coreos-maipo.spec [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Source0:        redhat-release-%{variant_lowercase}-%{base_release_version}-4.tar.gz
+
+%description
+%{summary}
+
+%prep
+%setup -q -n redhat-release-%{base_release_version}
+
+%build
+make fs-maipo
```
Comment (jlebon) [inline redhat-release-coreos-maipo.spec:31]: Heck, we could collapse this into a single spec and just use the `%rhel` macro from our buildroot, right?

**Comment by miabbott** [comment]
Will having two spec files cause a problem for `rdgo`?

**Comment by cgwalters** [comment]
> Will having two spec files cause a problem for rdgo?

Yep, it'll break it today.  I thought of that but had been putting off the problem :smile:  - But you're right to call this out now. Maybe the simplest is to use git branches for this, we could split off a `maipo` branch?

**Comment by miabbott** [comment]
> Maybe the simplest is to use git branches for this, we could split off a maipo branch?

or

> Heck, we could collapse this into a single spec and just use the %rhel macro from our buildroot, right?

I don't have enough experience with spec files to know if it would be easier to maintain two of them (porting changes between the two) or have a massive one with a bunch of conditionals.

I'd be fine with git branches for now, but defer to others with more expertise.

**Comment by cgwalters** [comment]
Obsoleted by https://github.com/openshift/redhat-release-coreos/pull/13

**Pull request closed without merging**